### ISSUE TITLE
BUG / Request: Improve speed of ListScap.do when displaying scap results to prevent timout issues.

### ISSUE THREAD

**Issue by digdilem** [uyuni-project/uyuni#6378]
https://{hostname}/rhn/systems/details/audit/ListScap.do?sid={serverid}

When there are several SCAP Audit results stored on this page (Audit tab from any server), the page loads extremely slowly, taking over 10 seconds, often up to 20 seconds for 5+ results.

This page is displayed whenever you open that tab and review the results. 

This delay also impacts scheduling new tests, as after submitting a new scheduled scan, it directs to this page. 

As other aspects of Uyuni have benefited from recent work on speeding them up, this is now even more apparent, and hinders the human flow.

Listing as an enhancement request rather than a bug, as the results do load eventually, but there may be a bug or inefficiency that is causing a slowness.

**Comment by digdilem** [comment]
This probably qualifies as a bug, rather than an enhancement. 

The delay in loading this page once there are more than a very few results, often causes Uyuni to timeout and makes it impossible to use the audit tab on a computer at all.  

Eg, this happens when trying to open the Audit tab when there are 34 sets of results.

![image](https://user-images.githubusercontent.com/4052275/227192857-94550667-4dfa-472a-aeb9-eaecd785bd2f.png)

Sometimes attempting to reload this same page will succeed, but it feels very much like there is a problem in the code that's causing exponentially increasing delays for every set of results. 

As it is, this aspect of Uyuni makes it unsuitable for Scap management, in my view.

**Comment by admd** [comment]
@digdilem we have removed the diff column with this PR https://github.com/uyuni-project/uyuni/pull/6876 . It wasn't adding much value but was very expensive in the end. Once this PR will be merged, the issue that you are facing should be resolved.

**Comment by digdilem-work** [comment]
@admd - Many thanks, look forwards to the new version.

**Comment by admd** [comment]
@digdilem current release should have the fix. Please check and if the problem is not there anymore, you can close this one.

**Comment by digdilem** [comment]
Hi @admd 

I missed the announcement about a new Uyuni release being available, so I have now upgraded that.

I did some retesting on these results before the upgrade.

Before, opening "Audit" tab on a vm with 40 test results:  Around 55 seconds. Refreshing tab was as slow. I didn't see the same timeout issue as above, which I guess happens at 60s, but that may be related to the vm load which was 0.10 for this test.

After the upgrade: Results showed **instantly**.

So yes - that is a significant improvement and I'll close with this. 

Thank you for the fix.
